src/hotspot/share/memory/metaspaceClosure.hpp

```diff
@@ -128,8 +128,8 @@ class MetaspaceClosure {
     virtual ~Ref() {}
 
     address obj() const {
-      // In some rare cases (see CPSlot in constantPool.hpp) we store some flags in the lowest
-      // 2 bits of a MetaspaceObj pointer. Unmask these when manipulating the pointer.
+      // In some rare cases we store some flags in the lowest 2 bits of a
+      // MetaspaceObj pointer. Unmask these when manipulating the pointer.
```

Comment:
I think `FLAG_MASK` and `flag_bits()` can be removed completely. It was used only for handling the CPSlot during CDS dump. I added the following and all CDS tests passed.

```
       uintx p = (uintx)*mpp();
+      assert((p & FLAG_MASK) == 0, "must be");
       return (address)(p & (~FLAG_MASK));
```

If you want, I can do the removal in a separate PR.

Reply:
I like the idea of doing this as a separate PR.

Reply:
I created https://bugs.openjdk.java.net/browse/JDK-8272343.